### ISSUE TITLE
[Feature Request] Please add a new category: Longest Supported Plugins

### ISSUE THREAD

**Issue by AB-boi** [ganesshkumar/obsidian-plugins-stats-ui#50]
**First, I would like to express my gratitude for your contribution to this resourceful website.**

## Feature Requested:
 I would like to suggest adding a new category to the website that lists the longest-supported plugins. 

## My Rationale:
Obsidian has seen a recent increase in downloads and new users, which has resulted in an increase in community contributions. The plugin store contains plugins of _varying quality and redundancy_. **A plugin's utility is directly proportional to its support.** 
Therefore, to maintain good practices and enhance the user experience, I think users should be able to view the most popular and longest-supported plugins that match their needs. 
Unfortunately, I am not an expert and cannot provide specific suggestions for designing the algorithm to accomplish this. 
If you agree with my point of view and think this is a worthwhile endeavor, I ask that you consider implementing this feature. 

**Thank you again for your time and consideration @ganesshkumar.**

**Comment by ganesshkumar** [comment]
Thanks for creating this. What do you mean by longest-supported? Are we trying to see if the plugin is being activly supported (new code commits, issues getting closed etc.) or we are also looking for how long an actively supported plugin has been in the plugin community? 

Identifying actively supported plugins is indeed a useful feature. Let me try to formulate how this can be achieved.

**Comment by AB-boi** [comment]
Yes, both the things you said can be included in the parameters. 
Plugins with:
- Active Github repos:
    - Updates frequently
    - Responsive to issues and the swiftness of their resolution
    - Responsiveness to feature requests
    - Recency of updates limit (must be updated last within an year)
- Time since publication is also useful to gauge the consistency of the plugin's support
- Ratio of Open to Close Issues
- I think you'll have to create a heirarchy of which parameters hold a higher priority so that those can be applied as the first filter.
- After you have filtered all the plugins, you can perhaps apply a sorting algo to display in the descending order of downloads.

I also got some input from **ChatGPT**:
```
### Parameters for Sorting Plugins (Descending Order of Priority):  

1. **Active Support Metrics**  
   - **Issue Resolution Rate**: Percentage of issues resolved within a defined timeframe (e.g., last 6 months).  
   - **Feature Request Response Rate**: Frequency of responses to feature requests, indicating ongoing support.  

2. **Recent Update Frequency**  
   - Timestamp of the **last update** (e.g., most recent code commit or release).  
   - Consistency of updates over time (e.g., number of updates in the past year).  

3. **Support Longevity**  
   - Total time since the plugin's first release and initial support date.  

4. **Compatibility with Latest Software Version**  
   - Verified compatibility with the current version of the core application (e.g., Obsidian).  

5. **Download Popularity (Secondary Priority for Relevance)**  
   - Number of downloads over a specified period (e.g., last month or year).  
   - Growth trend in downloads, showing rising or steady interest.  

6. **User Ratings and Reviews**  
   - Average user rating and the number of reviews to validate the plugin’s quality and reliability.  

7. **Documentation Completeness**  
   - Availability and clarity of setup guides, use cases, and troubleshooting tips.  

8. **Community Endorsements**  
   - Mentions or recommendations by trusted community moderators or users.  

This list prioritizes **active support** and **age of continuous support** to identify plugins that are both reliable and sustained over time. The sorting algorithm could rank plugins using a weighted score based on these parameters, emphasizing the top criteria more heavily.
```

While I was typing this I also got a couple of ideas for other categories:
- Fraudulent/Suspicious plugins (If there are any)(I am not sure if this can be done)
- Non-redundant plugins (minimal feature overlap/uniqueness)(Not sure if this can be done either)
(Perhaps these two would need manual work, yeah?)

**Comment by ganesshkumar** [comment]
As a first step, I have collected some metrics about the repository using Github API (stayed within API limits) and added them to the plugin view. 

I have collected count of stars, forks, issues (open, close), pull requests (open, merged, closed), days elapsed since plugin creation, last commit, last release, number of code commits in last one year (proportional to number of releases?).

![image](https://github.com/user-attachments/assets/cbe0c666-f100-4291-bbc8-bd8ac4841328)

Next steps:
- Fix the inconsistency between the plugin creation time, last commit time and last release time. They are showing wrong numbers as of now.
- Formulate weights to these to assign a score from 0-100
- Iterate :)

Note:
- I am not including issue or pull request average velocity (time between open and close/merge)

**Comment by ganesshkumar** [comment]
Started a discussion to keep the discussion about score beyond the life time of this issue https://github.com/ganesshkumar/obsidian-plugins-stats-ui/discussions/52

**Comment by ganesshkumar** [comment]
@AB-boi an early version of score is available now on the website, but I will be tweaking over the next few days before marking it as ready

**Comment by AB-boi** [comment]
That's nice!
